src/Equations/damaged-elastic/Kernels/Time.cpp

```diff
@@ -703,3 +700,1491 @@ void seissol::kernels::Time::flopsTaylorExpansion(long long& nonZeroFlops,
 }
 
 unsigned int* seissol::kernels::Time::getDerivativesOffsets() { return m_derivativesOffsets; }
+
+void seissol::kernels::Time::calculateEps(
+    const real* exxNodal,
+    const real* eyyNodal,
+    const real* ezzNodal,
+    const real* exyNodal,
+    const real* eyzNodal,
+    const real* ezxNodal,
+    const unsigned int& q,
+    const seissol::initializer::parameters::DamagedElasticParameters& damagedElasticParameters,
+    real& EspI,
+    real& EspII,
+    real& xi) {
+  const real epsInitxx = damagedElasticParameters.epsInitxx;
+  const real epsInityy = damagedElasticParameters.epsInityy;
+  const real epsInitzz = damagedElasticParameters.epsInitzz;
+  const real epsInitxy = damagedElasticParameters.epsInitxy;
+  const real epsInityz = damagedElasticParameters.epsInitxx;
+  const real epsInitzx = damagedElasticParameters.epsInitxx;
+
+  EspI = (exxNodal[q] + epsInitxx) + (eyyNodal[q] + epsInityy) + (ezzNodal[q] + epsInitzz);
+  EspII = (exxNodal[q] + epsInitxx) * (exxNodal[q] + epsInitxx) +
+          (eyyNodal[q] + epsInityy) * (eyyNodal[q] + epsInityy) +
+          (ezzNodal[q] + epsInitzz) * (ezzNodal[q] + epsInitzz) +
+          2 * (exyNodal[q] + epsInitxy) * (exyNodal[q] + epsInitxy) +
+          2 * (eyzNodal[q] + epsInityz) * (eyzNodal[q] + epsInityz) +
+          2 * (ezxNodal[q] + epsInitzx) * (ezxNodal[q] + epsInitzx);
+  if (EspII > 1e-30) {
+    xi = EspI / std::sqrt(EspII);
+  } else {
+    xi = 0.0;
+  }
+}
+
+void seissol::kernels::Time::computeNonLinearRusanovFlux(
```

Comment:
I guess technically, this would be better suited to be place in the `Local` kernel.

Reply:
Modified